Automattic/Gravatar-SDK-Android
# QuickEditor: Implement GravatarQuickEditorActivity

Closes #424 

### Description

To simplify the integration we need to remove the limitation of using the QuickEditor from an activity using the launchMode="singleTask".

The approach is to create an intermediate activity that can assume the complexity and the launch mode required by the QE. The behavior should be as much as possible as it is using the Composable directly.

https://github.com/user-attachments/assets/a0b8c091-ee85-4cf7-8e49-520f8b090d11

### Testing Steps

* In the `demo-app` manifest uncomment the `Gravatar QE Activity` part and comment the intent filter in `MainActivity` (or apply this [patch](https://github.com/user-attachments/files/17733845/Gravatar-Android-15-17-02.patch))
* Launch the `demo-app`
* Tap over `Test with Activity without Compose`
* Tap over the second button, `Update Avatar with QuickEditor Activity`
* Verify everything works as expected. 
  - OAuth flow with different browsers and using custom tabs (Firefox would be nice)
  - Avatar switch (Note that with this approach, we only notify the avatar change when our activity finishes)
  - Close the bottom sheet without switching avatars
  - You can also test rotating the device orientation with the bottom sheet (GravatarQuickEditorActivity) visible
